packages/firestore/src/lite-api/aggregate_types.ts

```diff
@@ -15,24 +15,45 @@
  * limitations under the License.
  */
 
+import { AggregateType } from '../core/aggregate';
+import { ObjectValue } from '../model/object_value';
+import { FieldPath as InternalFieldPath } from '../model/path';
+
 import { Query } from './reference';
+import { AbstractUserDataWriter } from './user_data_writer';
+
+export { AggregateType };
 
 /**
  * Represents an aggregation that can be performed by Firestore.
  */
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 export class AggregateField<T> {
   /** A type string to uniquely identify instances of this class. */
-  type = 'AggregateField';
+  readonly type = 'AggregateField';
+
+  /**
+   * Create a new AggregateField<T>
+   * @param _aggregateType Specifies the type of aggregation operation to perform.
+   * @param _internalFieldPath Optionally specifies the field that is aggregated.
+   * @internal
+   */
+  constructor(
+    // TODO (sum/avg) make aggregateType public when the feature is supported
+    readonly _aggregateType: AggregateType = 'count',
+    readonly _internalFieldPath?: InternalFieldPath
+  ) {}
 }
 
 /**
  * The union of all `AggregateField` types that are supported by Firestore.
  */
-export type AggregateFieldType = AggregateField<number>;
+export type AggregateFieldType =
+  | AggregateField<number>
+  | AggregateField<number | null>;
```

Comment:
are we trying to avoid a breaking change (`export type AggregateFieldType = AggregateField<number | null>`)?

Reply:
Right...well the definition was supposed to be `type AggregateFieldType = ReturnType<typeof count> | ReturnType<typeof sum> | ReturnType<typeof average>` and just tweaked this last minute when removing the feature from the public API.  I simplified this for now, but I've also added a TODO to return the code to the original clean proposal.